packages/x-data-grid/src/models/gridSlotsComponentsProps.ts

```diff
@@ -36,6 +35,18 @@ import type { GridColumnHeaderSortIconProps } from '../components/columnHeaders/
 
 type DividerProps = {};
 
+type MenuItemProps = {
+  autoFocus?: boolean;
+  children: React.ReactNode;
+  disableRipple?: boolean; // FIXME: consider removing these
```

Comment:
Fine-tuning elements case-by-case will become much harder with a design-system agnostic datagrid.

The only place where this is used is [here](https://github.com/mui/mui-x/blob/a1e7669293be64b33578e3cca6657350c3f46ff5/packages/x-data-grid-premium/src/components/GridColumnMenuAggregationItem.tsx#L69), any objection to removing it? cc @MBilalShafi, this was added in https://github.com/mui/mui-x/pull/6619

Reply:
I can see how the UX is weird when using the inner select. I'm not sure what would be the best way to solve the issue while removing material-specific vocabulary. One way to go would be to consider it a material-ui issue and push it to them to solve interactive components inside a menu item. Another would be to rename it to something more generic. Any thoughts/ideas?

Reply:
I've renamed that prop to `inert` and added jsdocs, we can iterate on that if we find a better solution.

Reply:
Would it be possible to have slotProps typed based on the design system?

Each slot could have a set of props that all design systems have to conform to:

```ts
type MenuItemProps = {
  autoFocus?: boolean;
  children: React.ReactNode;
  value: string
  disabled?: boolean;
  onClick?: React.MouseEventHandler<HTMLElement>;
  selected?: boolean;
  value?: number | string | readonly string[];
}
```

And then for the material version of the grid, the props could be typed as:

```ts
type MaterialMenuItemProps = MenuItemProps & MUIMenuItem
```

Otherwise, we might be degrading the ability to customize slots via slotProps 🤔